port/linuxaarch64/omrsignal_context.c

```diff
@@ -0,0 +1,319 @@
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#define _GNU_SOURCE
+
+#include <sys/ucontext.h>
+#include <unistd.h>
+
+#include "omrport.h"
+#include "omrsignal_context.h"
+
+#define NGPRS 32
+#define NFPRS 32
+
+/* AArch64 Signal Context Reserved Space Size */
+#define RESERVED_SPACE_SZ 4096
+
+void
+fillInUnixSignalInfo(struct OMRPortLibrary *portLibrary, void *contextInfo, struct OMRUnixSignalInfo *j9Info)
+{
+	j9Info->platformSignalInfo.context = (ucontext_t *)contextInfo;
+	/* module info is filled on demand */
+}
+
+uint32_t
+infoForSignal(struct OMRPortLibrary *portLibrary, struct OMRUnixSignalInfo *info, int32_t index, const char **name, void **value)
+{
+	*name = "";
+
+	switch (index) {
+
+	case OMRPORT_SIG_SIGNAL_TYPE:
+	case 0:
+		*name = "J9Generic_Signal_Number";
+		*value = &info->portLibrarySignalType;
+		return OMRPORT_SIG_VALUE_32;
+
+	case OMRPORT_SIG_SIGNAL_PLATFORM_SIGNAL_TYPE:
+	case 1:
+		*name = "Signal_Number";
+		*value = &info->sigInfo->si_signo;
+		return OMRPORT_SIG_VALUE_32;
+
+	case OMRPORT_SIG_SIGNAL_ERROR_VALUE:
+	case 2:
+		*name = "Error_Value";
+		*value = &info->sigInfo->si_errno;
+		return OMRPORT_SIG_VALUE_32;
+
+	case OMRPORT_SIG_SIGNAL_CODE:
+	case 3:
+		*name = "Signal_Code";
+		*value = &info->sigInfo->si_code;
+		return OMRPORT_SIG_VALUE_32;
+
+	case OMRPORT_SIG_SIGNAL_HANDLER:
+	case 4:
+		*name = "Handler1";
+		*value = &info->handlerAddress;
+		return OMRPORT_SIG_VALUE_ADDRESS;
+
+	case 5:
+		*name = "Handler2";
+		*value = &info->handlerAddress2;
+		return OMRPORT_SIG_VALUE_ADDRESS;
+	
+	case OMRPORT_SIG_SIGNAL_INACCESSIBLE_ADDRESS:
+	case 6:
+		/* si_code > 0 indicates that the signal was generated by the kernel */
+		if (info->sigInfo->si_code > 0) {
+			if ((info->sigInfo->si_signo == SIGBUS) || (info->sigInfo->si_signo == SIGSEGV)) {
+				*name = "InaccessibleAddress";
+				*value = &info->sigInfo->si_addr;
+				return OMRPORT_SIG_VALUE_ADDRESS;
+			}
+		}
+		return OMRPORT_SIG_VALUE_UNDEFINED;
+
+	// case OMRPORT_SIG_SIGNAL_ADDRESS:
+	// case 7:
+	// 	*name = "fault_address";
+	// 	*value = &((struct sigcontext *)&info->platformSignalInfo.context->uc_mcontext)->fault_address;
+	// 	return OMRPORT_SIG_VALUE_ADDRESS;
+
+
+	default:
+		return OMRPORT_SIG_VALUE_UNDEFINED;
+	}
+}
+
+
+/**
+ * Walk the 4k reserve space to get the floating point registers since the layout differ per-chip
+ */
+void*
+walkReserveSpace(uint32_t magic_to_find, uint8_t *reserved_space)
+{
+	if(!reserved_space) {
+		return NULL;
+	}
+	
+	int i=0;
```

Comment:
According to the coding standard https://github.com/eclipse/omr/blob/master/doc/CodingStandard.md:
Missing space before and after "=". Similarly space missing after "while" and "if" in the following lines.